cannot build inkscape on osx

## Issue description

nix-env -i inkscape-0.91 failed on osx 10.9.5

### Steps to reproduce
```
$ nix-env -i inkscape-0.91
...
In file included from context_surface_quartz.cc:21:
In file included from ../cairomm/quartz_surface.h:25:
/nix/store/v0l98kv189gk6c6d714rm0zc1gxgfhrm-cairo-1.14.6-dev/include/cairo/cairo-quartz.h:43:10: fatal error: 'ApplicationServices/ApplicationServices.h' file not found
#include <ApplicationServices/ApplicationServices.h>
         ^
1 error generated.
Makefile:561: recipe for target 'context_surface_quartz.lo' failed
make[2]: *** [context_surface_quartz.lo] Error 1
make[2]: Leaving directory '/private/var/folders/cy/4988f14n4r35fp39wrt1pfq80000gn/T/nix-build-cairomm-1.12.0.drv-0/cairomm-1.12.0/cairomm'
Makefile:622: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/private/var/folders/cy/4988f14n4r35fp39wrt1pfq80000gn/T/nix-build-cairomm-1.12.0.drv-0/cairomm-1.12.0'
Makefile:472: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/qc7v83m39siikalwwhip8z8s0yca6r2w-cairomm-1.12.0.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/w23jjgbqa3f5nnik8m8322v5r575828y-gtkmm-2.24.4.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/gvcg3brsj9zxda6fjwnvdxzxjdaqjlrm-python-2.7.11-bootstrapped-pip-8.1.2’
killing process 71136
cannot build derivation ‘/nix/store/5zb2z0hgy71jvybbvadqx5rqcj4jbg17-inkscape-0.91.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/5zb2z0hgy71jvybbvadqx5rqcj4jbg17-inkscape-0.91.drv’ failed
$
```
## Technical details

* System: Darwin Kernel Version 13.4.0: Mon Jan 11 18:17:34 PST 2016; root:xnu-2422.115.15~1/RELEASE_X86_64 x86_64
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: "16.09pre83147.df89584"

Progress, but libman.la won't link.

```
  CCLD     libman.la
Undefined symbols for architecture x86_64:
  "_program_name", referenced from:
      _error in libgnu.a(error.o)
      _error_at_line in libgnu.a(error.o)
ld: symbol(s) not found for architecture x86_64
clang-3.7: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:1526: recipe for target 'libman.la' failed
make[2]: *** [libman.la] Error 1
make[2]: Leaving directory '/private/var/folders/cy/4988f14n4r35fp39wrt1pfq80000gn/T/nix-build-man-db-2.7.5.drv-0/man-db-2.7.5/lib'
Makefile:1521: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/private/var/folders/cy/4988f14n4r35fp39wrt1pfq80000gn/T/nix-build-man-db-2.7.5.drv-0/man-db-2.7.5'
Makefile:1449: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/i5xrpqpvc71w5cj1nazi20clv78wv5qr-man-db-2.7.5.drv’ failed with exit code 2
building path(s) ‘/nix/store/8mk7wpiis4y9dicxdcgd0dcl1iihj7ma-openexr-2.2.0’, ‘/nix/store/plnhv7n1qv10qc49nyn3j5kf6caxhp05-openexr-2.2.0-doc’
cannot build derivation ‘/nix/store/4zf5amwgg7ajplicbihmsr89s99hblip-w3m-0.5.3-2015-12-20.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/bipny8ii8i8yd8qap5013nzj406mf579-xmlto-0.0.26.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/9q0922rh6w2l8s86af5przsy9mpwd2w3-giflib-5.1.0.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/5krnvl1ksrn17ccv6vq0kzg8ma657dz3-libwebp-0.4.3.drv’: 1 dependencies couldn't be built
killing process 82305
cannot build derivation ‘/nix/store/7yaih8szq9j7ylm3ngj6nc4dbdhsfksf-imagemagick-6.9.3-9.drv’: 1 dependencies couldn't be built
building path(s) ‘/nix/store/lqr1gpmr31gq52nrd0f89f7d8h3aciwz-openjpeg-2.1.0-dev’, ‘/nix/store/xkqqp8sp3s8pa5xm291jka6sjb0bda5h-openjpeg-2.1.0’
killing process 82307
cannot build derivation ‘/nix/store/4gffvfamx95rdb1cn001rx4ry75ksplr-inkscape-0.91.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/4gffvfamx95rdb1cn001rx4ry75ksplr-inkscape-0.91.drv’ failed
```

--- a/pkgs/applications/graphics/inkscape/default.nix
+++ b/pkgs/applications/graphics/inkscape/default.nix
@@ -1,8 +1,8 @@
-{ stdenv, fetchurl, pkgconfig, perl, perlXMLParser, gtk, libXft
+{ stdenv, fetchurl, fetchpatch, pkgconfig, perl, perlXMLParser, gtk, libXft
 , libpng, zlib, popt, boehmgc, libxml2, libxslt, glib, gtkmm
 , glibmm, libsigcxx, lcms, boost, gettext, makeWrapper, intltool
 , gsl, python, numpy, pyxml, lxml, poppler, imagemagick, libwpg, librevenge
-, libvisio, libcdr, libexif, unzip
+, libvisio, libcdr, libexif, unzip, automake114x, autoconf
 , boxMakerPlugin ? false # boxmaker plugin
 }:
 
@@ -12,6 +12,11 @@ boxmaker = fetchurl {
   # http://www.inkscapeforum.com/viewtopic.php?f=11&t=10403
   url = "http://www.keppel.demon.co.uk/111000/files/BoxMaker0.91.zip";
   sha256 = "5c5697f43dc3a95468f61f479cb50b7e2b93379a1729abf19e4040ac9f43a1a8";
+};
+
+stdcxx-patch = fetchpatch {
+  url = http://bazaar.launchpad.net/~inkscape.dev/inkscape/trunk/diff/14542?context=3;
+  sha256 = "15h831lsh61ichgdygkdkbdm1dlb9mhprldq27hkx2472lcnyx6y";
 };
 
 in
@@ -28,6 +33,7 @@ stdenv.mkDerivation rec {
   patches = [ ./deprecated-scopedptr.patch ];
 
   postPatch = ''
+    patch -i ${stdcxx-patch} -p 0
     patchShebangs share/extensions
   ''
   # Clang gets misdetected, so hardcode the right answer
@@ -46,7 +52,7 @@ stdenv.mkDerivation rec {
     pkgconfig perl perlXMLParser gtk libXft libpng zlib popt boehmgc
     libxml2 libxslt glib gtkmm glibmm libsigcxx lcms boost gettext
     makeWrapper intltool gsl poppler imagemagick libwpg librevenge
-    libvisio libcdr libexif
+    libvisio libcdr libexif automake114x autoconf
   ] ++ stdenv.lib.optional boxMakerPlugin unzip;
 
   enableParallelBuilding = true;

--- a/pkgs/development/libraries/cairomm/default.nix
+++ b/pkgs/development/libraries/cairomm/default.nix
@@ -1,4 +1,4 @@
-{ fetchurl, stdenv, pkgconfig, cairo, xlibsWrapper, fontconfig, freetype, libsigcxx }:
+{ fetchurl, stdenv, pkgconfig, darwin, cairo, xlibsWrapper, fontconfig, freetype, libsigcxx }:
 let
   ver_maj = "1.12";
   ver_min = "0";
@@ -13,7 +13,10 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ pkgconfig ];
   propagatedBuildInputs = [ cairo libsigcxx ];
-  buildInputs = [ fontconfig freetype ];
+  buildInputs = [ fontconfig freetype ]
+  ++ stdenv.lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+    ApplicationServices
+  ]);
 
   doCheck = true;
 
@@ -35,6 +38,6 @@ stdenv.mkDerivation rec {
     homepage = http://cairographics.org/;
 
     license = with licenses; [ lgpl2Plus mpl10 ];
-    platforms = platforms.linux;
+    platforms = platforms.unix;
   };
 }

